packages/xstate-react/src/createActorContext.ts

```diff
@@ -30,7 +30,7 @@ export function createActorContext<TLogic extends AnyActorLogic>(
     children,
     logic: providedLogic = actorLogic,
     machine,
-    options: providedOptions = interpreterOptions
+    options: providedOptions = actorOptions
```

Comment:
When we don't remove this default we often end up merging object A with object A (this becomes `{ ...actorOptions, ...actorOptions }`) and that bothers me slightly ;p
```suggestion
    options: providedOptions
```

We'll still be creating somewhat redundant copies of those options here since we spread them unconditionally but that should be fine an I'm OK with it.

To avoid those redundant copies we could have a `merge` util like:
```ts
function merge<T>(a: T | undefined, b: T | undefined): T | undefined {
  if (a === undefined) {
    return b;
  }
  if (b === undefined) {
    return a;
  }
  return {
    ...a,
    ...b,
  };

```